src/extensions/replay/sessionrecording.ts

```diff
@@ -906,11 +964,19 @@ export class SessionRecording {
             this._pageViewFallBack()
         }
 
+        // Check if the URL matches any trigger patterns
+        this._checkUrlTrigger()
```

Comment:
we _could_ check this only when the URL changes (we have a hook in this class already)

or return early in `checkUrlTrigger` if there's no reason to run the check

wandering towards early optimisation though - not blocking at all